File: cse241-object-oriented-programming/hw02/src/driver2.cpp
#include"tetris.h"

int main(){
    srand(time(0));
    int x,y;
    int endctr=0;
    cout << "Please enter the dimensions of the board (first input column, second input row)"<<endl;
    cin >> x;
    cin >> y;

    cout << "Game starting.."<<endl;
    sleep(1);

    Tetris mytetris(x,y);
    mytetris.initBoard();

    while(endctr <= (x-2)){
        char tetroinp;
        cout << "Tetromino type (I, O, T, J, L, S, Z).You may enter R for random Tetromino, Q for quit. : ";
        cin >> tetroinp;
        if(tetroinp == 'Q'){
            cout <<"Exiting..."<<endl;
            sleep(1);
            return 1;
        }
        else if(tetroinp != 'I' && tetroinp != 'O' && tetroinp != 'T' && tetroinp != 'J' && tetroinp != 'L' && tetroinp != 'S' && tetroinp != 'Z' && tetroinp != 'R' ){
            cout << "Wrong input! please try again."<<endl;
            continue;
        }
        if(tetroinp == 'R'){
            switch(rand() % 7){
                case 0:
                    tetroinp = 'I';
                    break;
                case 1:
                    tetroinp = 'O';
                    break;
                case 2:
                    tetroinp = 'T';
                    break;
                case 3:
                    tetroinp = 'J';
                    break;
                case 4:
                    tetroinp = 'L';
                    break;
                case 5:
                    tetroinp = 'S';
                    break;
                case 6:
                    tetroinp = 'Z';
                    break;
            }
        }
        mytetris.Add(tetroinp);
        if(mytetris.gettetrosvector()[(mytetris.gettotal())-1].getvector()[0].size() + mytetris.getbottomc() > x-1){
            break;
        }
        mytetris.Animate(mytetris.gettetrosvector());
    }
    cout << "Game Over"<<endl;
}

File: cse241-object-oriented-programming/hw07/src/driver2.java
import java.util.Scanner;
import java.util.Random;

public class driver2 {
    public static void main(String args[]){
        

        int x,y;
        int endctr = 0;
        Scanner input = new Scanner(System.in);
        do{
            System.out.print("Please enter the dimensions of the board (first input column, second input row)");
            x = input.nextInt();
            y = input.nextInt();
        }while(x<0 || y<0 );

        System.out.println("Game starting..");
        try { // for sleep 2 sec
            Thread.sleep(2000);
        } 
        catch (InterruptedException e) {
            e.printStackTrace();
        }

        Tetris mytetris = new Tetris(x,y);
        mytetris.Draw();

        while(endctr <= (x-2)){
            char tetroinp;
            System.out.println("Tetromino type (I, O, T, J, L, S, Z).You may enter R for random Tetromino, Q for quit. : ");
            tetroinp =  input.next().charAt(0);
            if(tetroinp == 'Q')
                return;

            else if(tetroinp != 'I' && tetroinp != 'O' && tetroinp != 'T' && tetroinp != 'J' && tetroinp != 'L' && tetroinp != 'S' && tetroinp != 'Z' && tetroinp != 'R'){
                System.out.println("Wrong input please try again.");
                continue;
            }
            if(tetroinp == 'R'){
                Random rand = new Random();
                char[] letters = {'I', 'O', 'T', 'J', 'S', 'L', 'Z'};
                int randomIndex = rand.nextInt(letters.length);
                char randomChar = letters[randomIndex];
                tetroinp = randomChar;
            }
            mytetris.Add(tetroinp);
            if(mytetris.gettetrosvector().get(mytetris.gettotal()-1).getcol() + mytetris.getbottomc() > x-1)
                break;
            mytetris.Animate();
        }
        System.out.println("Game over..");
    }
}
